src/java/populator/MapPopulator.java

```diff
@@ -300,12 +361,57 @@ public void populate() {
             slopesTexture.init(slopes, 0, 1).smooth(8).add(slopes, .65f).smooth(4).add(slopes, .5f).smooth(1).clampMax(1f);
             accentSlopesTexture.init(accentSlopes, 0, 1).smooth(8).add(accentSlopes, .65f).smooth(4).add(accentSlopes, .5f).smooth(1).clampMax(1f);
-            rockBaseTexture.init(rockBase, 0, 1).smooth(8).clampMax(0.35f).add(rockBase, .65f).smooth(4).clampMax(0.65f).add(rockBase, .5f).smooth(1).add(rockBase, 1f).clampMax(1f);
+            steepHillsTexture.init(steepHills, 0, 1).smooth(8).clampMax(0.35f).add(steepHills, .65f).smooth(4).clampMax(0.65f).add(steepHills, .5f).smooth(1).add(steepHills, 1f).clampMax(1f);
             rockTexture.init(rock, 0, 1).smooth(8).clampMax(0.2f).add(rock, .65f).smooth(4).clampMax(0.3f).add(rock, .5f).smooth(1).add(rock, 1f).clampMax(1f);
             accentRockTexture.init(accentRock, 0, 1).subtract(waterBeachTexture).clampMin(0).smooth(8).add(accentRock, .65f).smooth(4).add(accentRock, .5f).smooth(1).clampMax(1f);
 
+            if (keepLayer1) {
+                accentGroundTexture.replaceWith(oldLayer1);
+            }
+            if (keepLayer2) {
+                accentPlateauTexture.replaceWith(oldLayer2);
+            }
+            if (keepLayer3) {
+                slopesTexture.replaceWith(oldLayer3);
+            }
+            if (keepLayer4) {
+                accentSlopesTexture.replaceWith(oldLayer4);
+            }
+            if (keepLayer5) {
+                steepHillsTexture.replaceWith(oldLayer5);
+            }
+            if (keepLayer6) {
+                waterBeachTexture.replaceWith(oldLayer6);
+            }
+            if (keepLayer7) {
+                rockTexture.replaceWith(oldLayer7);
+            }
+            if (keepLayer8) {
+                accentRockTexture.replaceWith(oldLayer8);
+            }
+            if (keepLayer0) {
+                oldLayer0.subtract(oldLayer8).subtract(oldLayer7).subtract(oldLayer6).subtract(oldLayer5).subtract(oldLayer4).subtract(oldLayer3).subtract(oldLayer2).subtract(oldLayer1);
+                if (moveLayer0To == 1) {
+                    accentGroundTexture.replaceWith(oldLayer0);
+                } else if (moveLayer0To == 2) {
+                    accentPlateauTexture.replaceWith(oldLayer0);
+                } else if (moveLayer0To == 3) {
+                    slopesTexture.replaceWith(oldLayer0);
+                } else if (moveLayer0To == 4) {
+                    accentSlopesTexture.replaceWith(oldLayer0);
+                } else if (moveLayer0To == 5) {
+                    steepHillsTexture.replaceWith(oldLayer0);
+                } else if (moveLayer0To == 6) {
+                    waterBeachTexture.replaceWith(oldLayer0);
+                } else if (moveLayer0To == 7) {
+                    rockTexture.replaceWith(oldLayer0);
+                } else if (moveLayer0To == 8) {
+                    accentRockTexture.replaceWith(oldLayer0);
+                }
```

Comment:
Use a switch statement

Reply:
The replaceWith function is somewhat unnecessary as this could be done with maskName = new FloatMask(old, 0L, symmetryHierarchy)